app/ota_client.py

```diff
@@ -1170,19 +1183,24 @@ def _create_regular_file(
                     shutil.copy(reginf.path, dst)
                     processed.op = "copy"
                 else:
-                    processed.errors = downloader(
-                        reginf.path,
-                        dst,
-                        reginf.sha256hash,
-                    )
-                    processed.op = "download"
+                    # limit the concurrent downloading tasks
+                    with download_limiter:
+                        if True:
```

Comment:
Is this not required?

Reply:
This replaces the previous implementation for logic of limiting concurrent __downloading__ tasks.

Reply:
The outer semaphore(which the `_collector` is maintaining), is used to limit the number of total concurrent tasks(download, copy, link).

Reply:
> Is this not required?

Sorry, `if Ture:` is required?

Reply:
Oh I am sorry, I mistakenly left a debug used line, I will delete it now.